Framework/DataHandling/src/LoadILLReflectometry.cpp

```diff
@@ -396,15 +394,16 @@ void LoadILLReflectometry::loadDataIntoTheWorkSpace(
     for (size_t j = 0; j < m_numberOfPixelsPerTube; ++j) {
 
       // just copy the time binning axis to every spectra
-      m_localWorkspace->dataX(spec + nb_monitors) = m_localWorkspace->readX(0);
+      m_localWorkspace->setSharedX(spec + nb_monitors,
+                                   m_localWorkspace->sharedX(0));
 
       //// Assign Y
       int *data_p = &data(static_cast<int>(i), static_cast<int>(j), 0);
-      m_localWorkspace->dataY(spec + nb_monitors)
+      m_localWorkspace->mutableY(spec + nb_monitors)
           .assign(data_p, data_p + m_numberOfChannels);
 
       // Assign Error
-      MantidVec &E = m_localWorkspace->dataE(spec + nb_monitors);
+      auto &E = m_localWorkspace->mutableE(spec + nb_monitors);
       std::transform(data_p, data_p + m_numberOfChannels, E.begin(),
                      LoadHelper::calculateStandardError);
```

Comment:
See below, if this is just computing the `sqrt` of the counts then you can rewrite the whole block with a single `setHistogram` call.